package.json

```diff
@@ -133,7 +132,7 @@
     "cpy": "^7.0.1",
     "cross-env": "^7.0.2",
     "csv-validator": "^0.0.3",
-    "devtools-protocol": "0.0.805376",
+    "devtools-protocol": "0.0.863986",
```

Comment:
i only upgrades in the hopes that the type error from session.js would be resolved, but it wasn't. might as well keep it tho?

Reply:
Yeah, may as well. Similar change was made in https://github.com/GoogleChrome/lighthouse/pull/12199 but I don't know when we will be able to merge that (once timing issues are resolved).

Reply:
If pptr is going to be the new way of doing things, can we align with their `devtools-protocol`?  That should eliminate this error.

Reply:
That would put a limit on how soon we can adopt new protocol types: only as soon as Puppeteer does a release.

Reply:
I agree we can't let puppeteer types dictate when we update our version of the protocol, but maybe we can come up with a policy in the near future since we're hitching our wagon to puppeteer and we'll have to figure out an update cadence for that (and/or puppeteer-core) anyways (it makes sense to use its types since we'll use puppeteer as our protocol manager, but we're also deviating from puppeteer's [suggested Chrome version policy](https://github.com/puppeteer/puppeteer/#q-why-doesnt-puppeteer-vxxx-work-with-chromium-vyyy), so the protocol types don't necessarily need to be what puppeteer ships with).

For just this change, though:
- we should avoid suppressing an error if there's not much downside to avoiding the error
- the protocol definitions are huge, and typescript has to compile and check both versions. Looking at the `tsc` diagnostics, it's an extra 18k types and (in totally unscientific testing) an extra second of type check time just for the doubled protocol file and comparison in `session.js`.
- we're updating the protocol definitions for non-pressing reasons after six months of no issues, so one-month old definitions seem fine to use for now (they still include everything #12199 will need)

Reply:
> I agree we can't let puppeteer types dictate when we update our version of the protocol ...

> we should avoid suppressing an error if there's not much downside to avoiding the error

I don't understand how to take these two statements. If we drop devtools-protocol for puppeteer types, we are immediately tying ourselves to puppeteer. As soon as a change in the protocol comes in that we need (ex: the next Issue type), we are stuck unless Puppeteer updates in-sync.

If we want to move towards this we should ask Puppeteer if there's a way to make puppeteer use `devtools-protocol` directly, then we could override the transitive dep in our package.json.

Reply:
> > I agree we can't let puppeteer types dictate when we update our version of the protocol ...
> 
> > we should avoid suppressing an error if there's not much downside to avoiding the error
> 
> I don't understand how to take these two statements.

can take them like:

> I agree we can't let puppeteer types dictate when we update our version of the protocol, but maybe we can come up with a policy in the near future...
> For just this change, though: we should avoid suppressing an error if there's not much downside to avoiding the error

:D

I'm not suggesting we give up protocol decisions to puppeteer as a policy going forward, I'm suggesting we use `devtools-protocol@0.0.854822` for this update in particular since that's what puppeteer 8 is using, there's no downsides to using that version, and it avoids the `_session.send()` error for now.